From 89747834c6bff78103bd5a5ca74027aab89a302e Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Mon, 24 Mar 2025 22:01:01 +0000
Subject: [PATCH 4/6] Find BLAS with `find_package()`

Switches to `find_package()` insteaf of assuming that the reference BLAS
is available without any check. This also allows to use OpenBLAS.
---
 clients/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/clients/CMakeLists.txt b/clients/CMakeLists.txt
index 33b71d0d..12bd44d7 100644
--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -157,7 +157,8 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
         endif()
       endif()
     else()
-      set( BLAS_LIBRARY "blas" )
+      find_package( BLAS REQUIRED )
+      set( BLAS_LIBRARY "${BLAS_LIBRARIES}" )
     endif()
   else() # WIN32
     file(TO_CMAKE_PATH "C:/Program\ Files/AMD/AOCL-Windows/amd-blis/lib/ILP64/AOCL-LibBlis-Win-MT.lib" AOCL_BLAS_LIBRARY)
@@ -184,7 +185,7 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
     set( BLIS_CPP ../common/blis_interface.cpp )
   endif()
 
-  message(STATUS "Linking Reference BLAS LIB: ${BLAS_LIBRARY}")
+  message(STATUS "Linking BLAS LIB: ${BLAS_LIBRARY}")
 
   if ( WARN_NOT_ILP64_PREFERRED )
     message( WARNING "Using ${BLAS_LIBRARY} as reference library, 64-bit tests may fail. Test suite should be run with --gtest_filter=-*stress*")
-- 
2.43.0

